app/router.js

```diff
@@ -9,9 +9,11 @@ const router = express.Router();
 router.get('/', (req, res) => {
   return res.send(layout({ title: 'New Widget' }, form));
 })
+
 router.post('/', (req, res) => {
   const widget = new Widget(req.body);
-  widget.save();
+  validateForm(req, res, widget)
```

Comment:
can you move the validation to the model and have it be called on `save`?
It would look like this here:
```js
try {
  widget.save();
} catch (err) {
  // handle error
}
```

Reply:
Good catch. Will do.

Reply:
I've refactored all validations into the Model. 

Errors are handled in a try catch block now.